test/unit/EchoClientTest.php

```diff
@@ -181,6 +181,96 @@ function testGetEventsCsvFormat()
         $this->assertEquals($expectedResult,$result);
     }
 
+    function testGetEventsToCertainTime()
+    {
+        $this->setRequiredDefines();
+
+        $stubPersonaClient = $this->getMockBuilder('\Talis\Persona\Client\Tokens')
+            ->disableOriginalConstructor()
+            ->getMock();
+        $stubPersonaClient->expects($this->once())->method('obtainNewToken')->will($this->returnValue(array('access_token'=>'some-token')));
+
+        $expectedEvent = array("class"=>"test.expected.event");
+        $response = new \Guzzle\Http\Message\Response('200');
+        $response->setBody(json_encode(array("events"=>array(
+            $expectedEvent
+        ))));
+
+        $mockRequest = $this->getMock('\Guzzle\Http\Message\Request', array('send'), array('get',''));
+        $mockRequest->expects($this->once())->method('send')->will($this->returnValue($response));
+
+        $stubHttpClient = $this->getMock('\Guzzle\Http\Client', array('get'));
+        $stubHttpClient->expects($this->once())->method('get')->with('http://example.com:3002/1/events?limit=25&class=test.expected.event&key=foo&value=bar&from=2018-01-22T15%3A17%3A28%2B00%3A00')->will($this->returnValue($mockRequest));
```

Comment:
@malcyL is this the date format you are expecting?

Reply:
yes, is `2018-01-22T15:17:28+00:00` not the correct format? It is ISO.

Reply:
Echo is using the javascript validator isISO8601 method to check the validity of the parameter:

https://github.com/talis/echo-server/pull/120/files#diff-de77b05ed669485f2114729c25dfe2b3R31